Add discussion of package formats, expanding "Wheel vs Egg" discussion (#1397)
pypa/packaging.python.org: source/glossary.rst

    Egg

        A :term:`Built Distribution` format introduced by :ref:`setuptools`,

>         which has been replaced by :term:`Wheel`.  For details, see
>         :ref:`egg-format`.

Reviewer: What about something like this?

```suggestion
        which has been deprecated in favor of :term:`Wheel`.  For details, see
        :ref:`egg-format`.
```

Author: I personally prefer the stronger wording. PyPI does not accept eggs anymore, pip has dropped support for installing them and is soon even going to remove all support for detecting/listing/showing/uninstalling them (pypa/pip#12330). I'd like to avoid giving the impression that there is still some time to move away from eggs, because their last users really need to migrate now.

Reviewer: Indeed... works for me :)

(Consider this solved. I can not solve it. Has something really changed about this? I think I have seen others wondering why they can not resolve a thread when they expected to be able to.)
